### DISCUSSION TITLE
Question about NameFilter

### DISCUSSION THREAD

**Discussion by KKacer** [icsharpcode/SharpZipLib#478] [General]
First, congratulations for the great work,

Second, I have 2 questions which want to ask here:

Can you please provide a sample on using your NameFilter for creating a zip?
Let's imagine we want to Zip a whole folder containing some codes, I want to 
exclude any folder named "bin" and "obj" and their contents.

I had a successful build without using any filters, not too good with regex, and couldn't find examples about this topic for your product.

My used code:
```cs
var zip = new FastZip
{
    Password = ZipPass, CreateEmptyDirectories = true, UseZip64 = UseZip64.On
};
zip.CreateZip(OutputZipFullName, ZipInputDir, true, "");
```

In addition maybe an extra optional parameter named "exclude" to be implemented on FastZip, can be helpful.

----
My second question, which I think could be an easier one, and guess you already implemented it is: I also want to use AES 256 as the encryption of this zip file, how/where can I tell the SharpZipLib to use it as the encryption algorithm?

Regards.

**Comment by Numpsy** [comment]
There are a few examples of using FastZip with filters on the Wiki page at https://github.com/icsharpcode/SharpZipLib/wiki/FastZip#-using-the-filefilter-parameter. The string filter parameters on ```CreateZip``` can actually be a list of regular expressions rather than just a single one, and should support both including and excluding files.


As it stands now, FastZip unfortunately doesn't directly support creation of zip files with AES encryption - there is some discussion about that in #454 and #380 (you can however create AES encrypted zips using ```ZipOutputStream```.)

**Comment by KKacer** [comment]
Solved my issue with exclusions right after,
And about **AES**, yes I saw those 2 posts also,
And as you said I am already using **ZipOutputStream**,
but still didn't find where/how I should apply it.

**Comment by Numpsy** [comment]
If you're adding entries to ZipOutputStream and want to AES encrypt them, you can set the ```AESKeySize``` property of ```ZipEntry``` to 128 or 256 before calling ```PutNextEntry```

**Comment by Numpsy** [comment]
FastZip has now been extended with a new ```EntryEncryptionMethod``` property that enables it to directly create AES encrypted zips, and it also has new variants of ```CreateZip``` that take the filters as instances of ```IScanFilter``` rather than strings, which allow for greater flexibility in how filtering is done.
